pyomo/contrib/incidence_analysis/dulmage_mendelsohn.py

```diff
@@ -0,0 +1,60 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from pyomo.common.dependencies import networkx as nx
+from pyomo.contrib.incidence_analysis.common.dulmage_mendelsohn import (
+        dulmage_mendelsohn as dm_nx,
+        )
+
+def dulmage_mendelsohn(matrix_or_graph, top_nodes=None, matching=None):
+    """
+    COO matrix or NetworkX graph interface to the coarse Dulmage Mendelsohn
+    partition.
+    """
+    if isinstance(matrix_or_graph, nx.Graph):
+        # The purpose of handling graphs here is that if we construct NX graphs
+        # directly from Pyomo expressions, we can do away with our SciPy
+        # dependency (and associated overhead).
+        graph = matrix_or_graph
+        partition = dm_nx(graph, top_nodes=top_nodes, matching=matching)
+    else:
+        # Assume matrix_or_graph is a scipy coo_matrix
```

Comment:
Should you assert this assumption?

Reply:
Maybe, but we could want to support objects that are duck-typed to look like a `coo_matrix`